première implémentation du défi 1 par Triplek

KrohomKrohom opened the pull request.

KrohomKrohom committed:
parser

destinvianel committed:
first pull request

KrohomKrohom committed:
nom des fonctions + clair

KrohomKrohom committed:
Merge branch 'triplek' of github.com:yurug/pcomp-2019 into triplek

qaco reviewed (changes requested):
# Correction

* ne compile pas (erreurs de syntaxe)
* Erreurs de typage : ErrorCell n'étend pas Cell, signature de splitData...
* dans la génération des formules, les bornes des entiers ne sont pas vérifiées

# Lisibilité

* Nommage "franglais"
* Différents niveaux d'abstraction au même niveau d'indentation (root de la classe Parser)

# Simplicité

* blocs inutilement complexes (exemple : le for de construction des formules)
* ce code peut être considérablement réduit en utilisant le pattern matching
* fonctions très longues et niveau d'imbrication parfois important

# Robustesse

* try catch Exception -> impossible de distinguer les problèmes
* bugs rendus possibles par le fait que les traitements sont faits dans des variables mutables

# Extensibilité

* difficile d'étendre (sans réécrire) à d'autres types de formule que le comptage

# Modularité

* le parser a de nombreux points d'entrée : c'est un problème
* tout est statique (rien n'est paramétré par des attributs) donc il n'y a pas besoin d'une classe, un object suffit

KrohomKrohom committed:
fix d'erreurs de syntaxe

KrohomKrohom committed:
version qui marche (pas encore testée)

KrohomKrohom committed:
version fonctionnelle

KrohomKrohom committed:
correction de l'algo (liste inversée)

KrohomKrohom committed:
meilleur affichage

KrohomKrohom committed:
privatisation des méthodes de parser pour limiter les points d'entrée

KrohomKrohom committed:
vérification sur les bornes pour les données de type B

KrohomKrohom committed:
typo

Karnaj committed:
Functional version.

With some improvements :
- deals with big files (around one minute for CSV of 15 Mo),
- some operations could be parallelize.

TODO : improve error management, use a `Cell` to delete some responsibilities of `Change`.

Karnaj committed:
Adds ws file (executes the programm).

Karnaj committed:
Update Makefile

Karnaj committed:
Fix errors with AChange in user commands

In the previous code, the value was not changed, as `v` took the value `valueWithInitialA`.

Karnaj committed:
Create pincip.md

Karnaj committed:
Fix change adding in modifier

If there was no change at the position of `c` in the list of already applied changes, then when we add `c`, its oldValue is the value of this cell in the file.

Karnaj committed:
Update and rename pincip.md to algorithm_principle_and_structure.md

Karnaj committed:
Update printer.scala

Karnaj committed:
Update utils.scala

Karnaj committed:
Update main.scala

Karnaj committed:
Rename algorithm_principle_and_structure.md to README.md

KrohomKrohom committed:
changement des noms de fichier

KrohomKrohom committed:
fonctions anonymes

KrohomKrohom committed:
factorisation de Cell

Karnaj committed:
Update Change.scala

Karnaj committed:
Fix evaluate method.

When the `applyChange` method is called by `evaluate`, we should consider that the change which is evaluated is already viewed w.r.c the graph.

Karnaj committed:
Fix CSV Printer

Karnaj committed:
Fix propagation

Even if there is no change, we should apply (because we can found a cycle).

Karnaj committed:
Fix add of a new change.

Karnaj committed:
Update CSVPreprocessor.scala

Karnaj committed:
Update Change.scala

Karnaj committed:
Adds some tests.

Karnaj committed:
Make ws executable.

KrohomKrohom committed:
nomss de fonctions + parlant

Karnaj committed:
Fix bad commit for tests.

Karnaj committed:
Merge branch 'triplek' of github.com:yurug/pcomp-2019 into triplek

Karnaj committed:
Update CSVPreprocessor.scala

Karnaj committed:
A bit more efficient evaluation.

Karnaj committed:
More efficient preprocessing

Karnaj committed:
More efficient CSVParser

Deletes zipWithIndex and list concatenation.

Karnaj committed:
A bit more efficient CellParser

Karnaj committed:
A bit more efficient Printer.

Deletes zipWithIndex.

Karnaj committed:
Update Main.scala

Karnaj committed:
Update utils.scala

Karnaj committed:
Adds Dockerfile

Karnaj committed:
Fix printer.

Karnaj committed:
Update test.rb

Karnaj committed:
Better tests

Karnaj committed:
Merge branch 'triplek' of github.com:yurug/pcomp-2019 into triplek

Karnaj committed:
Don't print change when replace a formula by itself.

Karnaj committed:
Use CSVPreprocessor with List but without all improvements.

Karnaj committed:
Use sbt and improve algorithm.

Karnaj committed:
Update Dockerfile

Karnaj committed:
Evaluate the connected components concurently using Akka.

Karnaj committed:
Use auxiliary CSV to apply commands (useful if too many commands).

Karnaj committed:
Documentation.

Karnaj committed:
Documenta.

Karnaj committed:
Update Main.scala

Karnaj committed:
Update CSVPreprocessor.scala

Karnaj committed:
Update CSVPreprocessor.scala

Karnaj committed:
Update Evaluator.scala
